### ISSUE TITLE
Unable to suppress Quart serving logs after Python 3.10 upgrade

### ISSUE THREAD

**Issue by johndonor3** [pallets/quart#224]
I'm moving a Quart webapp from python 3.7 to python 3.10 and I'm suddenly unable to suppress the server logs.

I would expect `getLogger('quart.serving').setLevel(ERROR)`  to suppress most logging messages, but after switching to 3.10 I get everything.

Environment:

- Python version: 3.10.9
- Quart version: 0.18.3

**Comment by andrewshadura** [comment]
Since 0.18.0, [there isn’t](https://github.com/pallets/quart/commit/a54dab248714f08ff9f49acb32dd47834f86ddc1) a separate serving logger, that’s why this code no longer works.

**Comment by johndonor3** [comment]
Ah, yep. That would explain it. It's worth noting the [docs](https://quart.palletsprojects.com/en/latest/how_to_guides/logging.html) are out of date (part of my confusion).

But more importantly: it is not obvious to me how to recover the functionality I'm interested in. Previously I was able to easily suppress all of the serving logs (and since my app runs as a systemd daemon, I really don't want them, I'm only interested in seeing errors in the logs). Based on the comments in that commit, I thought maybe I could suppress the hypercorn logging: `getLogger('hypercorn.serving').setLevel(ERROR)` but no luck there.

**Comment by andrewshadura** [comment]
You can manually construct hypercorn's config and ask it to serve your Quart app, then you can configure logging as you wish. That's what I do, since my app needs to be served on multiple ports simultaneously, and I couldn't achieve what I wanted with plain Quart.

-- 
Cheers,
  Andrej

**Comment by davidism** [comment]
Hypercorn as used by `Quart.run` is a development server. In production, you should run Hypercorn directly, at which point you can configure it as needed.

**Comment by davidism** [comment]
Why don't you want them in development?

**Comment by johndonor3** [comment]
I am using the same app to serve some endpoints that are pinged every few seconds by other parts of the app. There is one important page that needs to refresh some tables and graphs in near real time so if I'm working on that page the logs are a constant stream of requests for those endpoints. 

This is probably not best practice? But I'm self-taught and the only developer for this project, and it's only one of my responsibilities.

Also "serving" is never the thing that's broken, haha. But if I knew how to suppress those logs, I could un-suppress them if I ever needed to.

**Comment by davidism** [comment]
> This is probably not best practice?

No, you probably want websockets, especially since you're using Quart so they're easily available.

**Comment by pwwang** [comment]
We are supposed to be able to customize loggers via `hypercorn.config.Config`:

```python
# define your own logger
logger = ...  

from hypercorn.config import Config

Config.accesslog = logger
Config.errorlog = logger
```

However, the following code from `quart` prevents us to do this:

https://github.com/pallets/quart/blob/c9768af63097c6ef3405cfd42d46b49ad6d1436d/src/quart/app.py#L1469-L1480

We should not overwrite the config directly but set the default values instead.

Something like:

```python
if not isinstance(config.accesslog, (logging.Logger, logging.LoggerAdapter)):
    config.accesslog = "-"
```

PS: We should not only allow `logging.Logger`, but also `logging.LoggerAdapter` here:

https://github.com/pgjones/hypercorn/blob/8ae17ca68204d9718389fb3649ca0ed6ba851906/src/hypercorn/logging.py#L31-L33

```python
    if isinstance(target, logging.Logger):
        return target
```

**Comment by davidism** [comment]
> Hypercorn as used by Quart.run is a development server. In production, you should run Hypercorn directly, at which point you can configure it as needed.

**Comment by pwwang** [comment]
@davidism I agree.

@johndonor3 Just in case you do want to customize your logger with Quart in development mode:

See if this helps:

https://github.com/pwwang/pipen-board/blob/2429a64746c1eb4a3ccf48c7ad20c42f43ca22ff/pipen_board/defaults.py#L38-L61

```python
class Quart(_Quart):

    def run_task(
        self,
        host: str = "127.0.0.1",
        port: int = 5000,
        debug: bool | None = None,
        ca_certs: str | None = None,
        certfile: str | None = None,
        keyfile: str | None = None,
        shutdown_trigger: Callable[..., Awaitable[None]] | None = None,
    ) -> Coroutine[None, None, None]:
        config = HyperConfig()
        config.access_log_format = "%(r)s %(s)s %(b)s %(D)s"
        config.accesslog = logger.logger
        config.bind = [f"{host}:{port}"]
        config.ca_certs = ca_certs
        config.certfile = certfile
        if debug is not None:
            self.debug = debug
        config.errorlog = config.accesslog
        config.keyfile = keyfile

        return serve(self, config, shutdown_trigger=shutdown_trigger)

```
